File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
def solution(n, m):
    a = [i for i in range(1, n+1) if n%i==0]
    b = [i for i in range(1, m+1) if m%i==0]
    
    common_values = [i for i in a if i in b]
    
    if len(common_values) == 1:
        gcd = 1
        lcm = n*m
    else:
        gcd = max(common_values)
        lcm = n*m//gcd
    
    return gcd, lcm

File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.js
function solution(n, m) {
    var answer = [];
    
    let a = Math.max(n, m);
    let b = Math.min(n, m);

    
    while(b!=0){
        let r = a % b
        a = b;
        b = r;
    }
    
    answer.push(a, parseInt((n*m)/a))
    
    return answer;
}